torch/csrc/autograd/engine.cpp

```diff
@@ -579,8 +579,7 @@ void GraphTask::exec_post_processing() {
     c10::MultiStreamGuard g(caller_current_streams_filtered);
     // WARNING: Don't use a range-for loop here because more callbacks may be
     // added in between callback calls, so iterators may become invalidated.
-    // NOLINTNEXTLINE(modernize-loop-convert)
-    for (size_t i = 0; i < final_callbacks_.size(); ++i) {
+    for (const auto i : c10::irange(final_callbacks_.size())) {
```

Comment:
`c10::irange()` captures upper boundary at the start of the loop, but here it can change (for example increase, as tested in `TestAutograd.test_callback_adds_callback`), so any form of range loop is really no go. Perhaps in order to get rid of modernize-loop warning, a `while` loop could have been used here